typhon/topography.py

```diff
@@ -0,0 +1,425 @@
+"""
+The topography module provides interfaces to global elevation models.
+So far, only an interface to the `SRTM30
+<https://dds.cr.usgs.gov/srtm/version2_1/SRTM30/srtm30_documentation.pdf>`_ data
+set is provided, which has a resolution of 1 km.
+
+Elevation data is downloaded on the fly but is cached to speed up subsequent
+access. The interfaces uses the path pointed to by the :code:`TYPHON_DATA_PATH`
+environment variable as data cache. This means that data is downloaded only
+when they are not found in the cache.
+
+.. note:: If :code:`TYPHON_DATA_PATH` is not set, a temporary directory will be
+    used and the data will be downloaded each time the module is used. This will be
+    highly inefficient, so for recurring use it is recommended to use the caching
+    functionality.
+
+The module can be used in two ways:
+
+ 1. by extracting the elevation data at native resolution
+ 2. by interpolating to elevation data at arbitrary locations
+
+The two different use cases are described below.
+
+Native resolution
+-----------------
+
+Extracting elevation data at native resolution for a given rectangular domain
+ is done using the :code:`SRTM30.elevation` function. The function returns a
+ tuple consisting of the latitude and longitude grids as well as the elevation
+ data in meters.
+
+.. code-block:: python
+
+    lat_min = 50
+    lon_min = 10
+    lat_max = 60
+    lon_max = 20
+    lats, lons, z = SRTM30.elevation(lat_min, lon_min, lat_max, lon_max)
+
+Interpolation to given coordinates
+----------------------------------
+
+Interpolation of the elevation data to arbitrary coordinates can be performed
+using the :code:`interpolate` method. Interpolation uses nearest neighbor
+interpolation and is implemented using a :code:`KDTree`.
+Interpolating the SRTM30 data to given latitude and longitude grids can be done
+as follows:
+
+.. code-block:: python
+
+    lat_min = 50
+    lon_min = 10
+    lat_max = 60
+    lon_max = 20
+    lats = np.linspace(lat_min, lat_max, 101)
+    lons = np.linspace(lon_min, lon_max, 101)
+    z = SRTM30.interpolate(lat, lont)
+"""
+import atexit
+import os
+import pickle
+import shutil
+import tempfile
+import urllib
+import zipfile
+
+import numpy as np
+
+import typhon
+from typhon.environment import environ
+
+clean_up_folders = []
+def _clean_up():
+    """
+    Function that takes care of cleaning up temporary directories.
+    """
+    for f in clean_up_folders:
+        shutil.rmtree(f)
+
+# Determine path to store tiles.
+try:
+    data_path = environ["TYPHON_DATA_PATH"]
+except:
+    data_path = tempfile.mkdtemp()
+    clean_up_folders += [data_path]
+    atexit.register(_clean_up)
```

Comment:
Since the tiles are not that small, it seems a bit reckless to download them every time again. Instead of storing them into a temporary directory, how about saving them in [$XDG_CACHE_HOME](https://wiki.archlinux.org/index.php/XDG_Base_Directory#Specification) (if the environment variable is set, otherwise default to ~/.cache/typhon)